Package@swift-4.2.swift

```diff
@@ -0,0 +1,67 @@
+// swift-tools-version:4.2
+/*
+ * Copyright 2017, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import PackageDescription
+import Foundation
+
+var packageDependencies: [Package.Dependency] = [
+  // Official SwiftProtobuf library, for [de]serializing data to send on the wire.
+  .package(url: "https://github.com/apple/swift-protobuf.git", .upToNextMajor(from: "1.5.0")),
+  
+  // Command line argument parser for our auxiliary command line tools.
+  .package(url: "https://github.com/kylef/Commander.git", .upToNextMinor(from: "0.8.0"))
+]
+
+var cGRPCDependencies: [Target.Dependency] = []
+
+if ProcessInfo.processInfo.environment.keys.contains("GRPC_USE_OPENSSL") {
+  packageDependencies.append(.package(url: "https://github.com/apple/swift-nio-ssl-support.git", from: "1.0.0"))
```

Comment:
This is actually a NIO dependency so it isn't necessary.

For the CgRPC dependencies, what we want is "if we're using linux or `GRPC_USE_OPENSSL` is set then don't depend on BoringSSL".

Reply:
I didn't manage to make it work without this NIO dependency (which actually is just libressl)
I've found @MrMage PR from some time ago regarding this, but even then this dependency was included: #238 
I've brought back `CFLAGS` used for linux .

Reply:
Oh sorry, I was completely wrong there, I didn't realise it was the support package.